### ISSUE TITLE
OboardingResponse#parseFrom causes NPE

### ISSUE THREAD

**Issue by original-codematrix** [DKE-Data/agrirouter-api-java#99]
**Describe the bug**
We use this library to interpret messages from the AgriRouter. With the current newest version we get a `NullPointerException`.

**To Reproduce**
_We use Kotlin_
We have an abstract class like this:
```kotlin
abstract class Foo<in K : GeneratedMessageV3?>(private val parser: ByteString.() -> K? = { null }) {...}
```
In there we have a method that gets a call from another class that executes the `parser`:
```kotlin
message. //Is a parameter from type K
    responsePayloadWrapper.
    details?.
    value?.
    parser() //Here the NullPointerException gets thrown
```
For this class, we have different implementations e. g.:
```kotlin
class Bar() : Foo<CloudVirtualizedAppRegistration.OnboardingResponse>(
	CloudVirtualizedAppRegistration.OnboardingResponse::parseFrom
) {...}
```

**Expected behavior**
No NullPointerException.

**Desktop (please complete the following information):**
 - Kotlin Version: 1.3.50
 - Library:
```
com.agrirouter.api.lmis:agrirouter-api-java-impl:0.0.1-SNAPSHOT
```
 - Java Version:
```bash
openjdk version "1.8.0_121"
OpenJDK Runtime Environment (IcedTea 3.3.0) (Alpine 8.121.13-r0)
OpenJDK 64-Bit Server VM (build 25.121-b13, mixed mode)
```

**Comment by saschadoemer** [comment]
Hey,

are you able to share a small example project with a test to reproduce the behaviour?

Regards

Sascha

P.S. Thanks for the details, thumbs up for a really great issue description!

**Comment by original-codematrix** [comment]
Hello @saschadoemer,

we already fixed this issue on our side. The problem here is/was the feature from Kotlin that allows you to pass static-references in an extension function through the constructor of the class. The fix here is to call the static `#parseForm`-method inside of the working method instead of passing a static reference through the constructor. Also, we switched from the old `0.0.1-SNAPSHOT` to the newest `1.0.0-jdk8-RC1` version.

I also need to say that the code above worked on my local machine fine but on our server-side, it causes the NPE but with our changes now the problem is solved. 

Regards,
Jan
